### PR TITLE
online dqm HLT objects monitoring module

### PR DESCRIPTION
**Pull request by BetterWang**
adding online DQM HLT objects monitoring module for 5 TeV pp ref run.

### CODE REVIEW

File: DQM/HLTEvF/python/HLTObjectMonitorPPReference_cff.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+from DQM.HLTEvF.HLTFullIterTrackingMonitoring_cff import *
+
+
+hlt4vector = cms.Path(
+    lumiOnlineMonitorHLTsequence # lumi
+    * hltObjectMonitor
+    * hcalOnlineMonitoringSequence # HCAL monitoring
+    * pixelOnlineMonitorHLTsequence # pixel cluster monitoring
+    * sistripOnlineMonitorHLTsequence # strip cluster monitoring
+    * trackingMonitoringHLTsequence # tracking monitoring
```
Comment (mtosi) [inline DQM/HLTEvF/python/HLTObjectMonitorPPReference_cff.py]: do you need it ?

File: DQM/HLTEvF/python/HLTObjectMonitorPPReference_cff.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+from DQM.HLTEvF.HLTObjectMonitorPPReference_cfi import *
+from DQM.HLTEvF.HLTFullIterTrackingMonitoring_cff import *
+
+
+hlt4vector = cms.Path(
+    lumiOnlineMonitorHLTsequence # lumi
+    * hltObjectMonitor
+    * hcalOnlineMonitoringSequence # HCAL monitoring
+    * pixelOnlineMonitorHLTsequence # pixel cluster monitoring
+    * sistripOnlineMonitorHLTsequence # strip cluster monitoring
```
Comment (mtosi) [inline DQM/HLTEvF/python/HLTObjectMonitorPPReference_cff.py]: I think you need to update the track collection name

File: DQM/HLTEvF/python/HLTObjectMonitorPPReference_cff.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+from DQM.HLTEvF.HLTObjectMonitorPPReference_cfi import *
+from DQM.HLTEvF.HLTFullIterTrackingMonitoring_cff import *
+
+
+hlt4vector = cms.Path(
+    lumiOnlineMonitorHLTsequence # lumi
+    * hltObjectMonitor
+    * hcalOnlineMonitoringSequence # HCAL monitoring
+    * pixelOnlineMonitorHLTsequence # pixel cluster monitoring
```
Comment (mtosi) [inline DQM/HLTEvF/python/HLTObjectMonitorPPReference_cff.py]: I think you need to update the track collection name

File: DQM/HLTEvF/python/HLTObjectMonitorPPReference_cff.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    * sistripOnlineMonitorHLTsequence # strip cluster monitoring
+    * trackingMonitoringHLTsequence # tracking monitoring
+    * egmTrackingMonitorHLTsequence # EGM tracking monitoring
+    * vertexingMonitorHLTsequence # vertexing
```
Comment (mtosi) [inline DQM/HLTEvF/python/HLTObjectMonitorPPReference_cff.py]: are you sure about those modules ?
do you need all of them ?
did you check the name of your collections ?

File: DQM/HLTEvF/python/HLTObjectMonitorPPReference_cff.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    * hltObjectsMonitor
```
Comment (mtosi) [inline DQM/HLTEvF/python/HLTObjectMonitorPPReference_cff.py]: do you need it ?

File: DQM/Integration/python/clients/hlt_dqm_sourceclient-live_cfg.py [hunk 6]
```diff
@@ -104,6 +104,9 @@
 ### for Proton-Lead collisions only (2016 Proton-Lead Era)
 #process.load("DQM.HLTEvF.HLTObjectMonitorProtonLead_cff")
 
+### for pp collisions 5 TeV pp reference run (2017 ppRef Era)
+#process.load("DQM.HLTEvF.HLTObjectMonitorPPReference_cff")
```
Comment (mtosi) [inline DQM/Integration/python/clients/hlt_dqm_sourceclient-live_cfg.py]: uhmmm, I think you need to have this line un-commented !!!

**Comment by mtosi** [comment]
@BetterWang thanks for this update
I think you need to 
1. make a backport for 92x, which I think will be the release used by online DQM 
@threus @dmitrijus can you confirm it, please ?
2. update some configuration files for the ppRef

**Comment by threus** [comment]
@mtosi @BetterWang yes, please make a backport to 92X (and base it on the 9_2_14 release)

**Comment by mtosi** [comment]
thanks for the prompt feedback !

File: DQM/HLTEvF/python/HLTObjectMonitorPPReference_cff.py (hunk 1 above)
Comment (BetterWang) [reply DQM/HLTEvF/python/HLTObjectMonitorPPReference_cff.py]: this 5 TeV pp ref run has all the standard pp sequences to monitor, on top of that, we added a few special objects.

File: DQM/HLTEvF/python/HLTObjectMonitorPPReference_cff.py (hunk 2 above)
Comment (BetterWang) [reply DQM/HLTEvF/python/HLTObjectMonitorPPReference_cff.py]: Could you be more specific? We should have all the standard pp sequence, but I might be wrong. I will check with out HLT expert.

**Comment by mtosi** [comment]
I see, I thought you will not run the standard pp HLT tracking

File: DQM/HLTEvF/python/HLTObjectMonitorPPReference_cff.py (hunk 4 above)
Comment (BetterWang) [reply DQM/HLTEvF/python/HLTObjectMonitorPPReference_cff.py]: We would like to keep the standard pp monitor, and add our stuff.
Why and which one do you think they are not needed?
 I will check with out HLT expert.

File: DQM/Integration/python/clients/hlt_dqm_sourceclient-live_cfg.py (hunk 6 above)
Comment (BetterWang) [reply DQM/Integration/python/clients/hlt_dqm_sourceclient-live_cfg.py]: Will un-comment in the back port.

**Comment by BetterWang** [comment]
Hi @mtosi, we will run standard pp HLT tracking, and, on top of that, full iteration tracking. So we would like to keep both stand pp monitor and our monitor. I will double check with our HLT expert.

**Comment by mtosi** [comment]
make sense, indeed, thanks !

**Comment by mtosi** [comment]
ah, I did not get it, sorry !

**Comment by dmitrijus** [comment]
@mtosi The release we use depends very closely on the HLT. If you we switch to 94x or later, we will switch too. So, yes, 92x :)

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
+1

**Comment by dmitrijus** [comment]
+1

Seeing that 92x is closed, should this be merged? It not, please ignore my signature and close this PR.

**Comment by BetterWang** [comment]
Hi @dmitrijus , this PR should be merged for future reference. The 92X PR is closed because the same PR #21222 is open to base on 9_2_14 for the onlineDQM P5TagCollector:
https://twiki.cern.ch/twiki/bin/viewauth/CMS/DQMP5TagCollector

Thanks,

**Comment by Martin-Grunewald** [comment]
@davidlange6 
Please merge this PR, or comment so it can be acted on.

**Comment by davidlange6** [comment]
sorry for my slow review - the proposed new module is another copy of an existing model with some variable names changed (maybe there are three plots add). Please take the time to make this configuration driven.

**Comment by Martin-Grunewald** [comment]
@BetterWang 
Please address the comments raised.

**Comment by BetterWang** [comment]
Hi @Martin-Grunewald @davidlange6 , due to a upcoming pre-approval, I will work on the PR next week. Sorry for the delay.

**Pull request closed without merging**